src/portable/portable.c

```diff
@@ -1300,7 +1300,7 @@ static int attach_unit_file(
                 if (fd < 0)
                         return log_debug_errno(fd, "Failed to create unit file '%s': %m", path);
 
-                r = copy_bytes(m->fd, fd, UINT64_MAX, COPY_REFLINK);
+                r = copy_bytes(m->fd, fd, UINT64_MAX, COPY_REFLINK|COPY_FSYNC);
```

Comment:
strictly speaking this will sync too early, as we will still chmod and will then move the file into place and never sync there...

I think it would be nicer to convert link_tmpfile()'s final parameter to a proper flag parameter, that takes the LINK_TMPFILE_REPLACE and LINK_TMPFILE_FSYNC parameters, and then does things properly, i.e. syncs the inode before moving it into place, and then does an fsync_full() once it moved it into place.

Reply:
i'll prep a patch for that